### PR TITLE
Respect keep_3d_linear when transparent_bg is on.

### LINKED ISSUE #26817: 3D viewport converted to sRGB despite keep_3d_linear in some cases.
**Godot version:** e930fb9 (3.1 beta11). Still present on 2940475 (master at the time of writing). GLES3 backend.

**OS/device including version:** Recent Arch Linux. GTX 970 w/ binary driver 418.43-4

**Issue description:** Viewport content converted to sRGB despite `keep_3d_linear`, when `transparent_bg` is on.

Bug is apparently caused by `RasterizerSceneGLES3::_post_process` disregarding `RENDER_TARGET_KEEP_3D_LINEAR` when `RENDER_TARGET_TRANSPARENT` is on or the render target is too small. Patch to illustrate the problem: https://github.com/godotengine/godot/compare/master...toasteater:fix/keep-linear

The patch fixes the bug in question for my use case, but as I'm not familiar with Godot's code base, I'm not sure if the change has any ramifications I'm not aware of, or if there is a better way to fix this problem. I would make a pull request if it's ok.

**Steps to reproduce:**
* Create GLES3 project.
* Add Node2D -> ViewportContainer -> Viewport
* Put something in Viewport. Turn on `keep_3d_linear` and `transparent_bg` for Viewport.

Expected: 3D content is in linear space.
Actual: 3D content converted to sRGB.

**Minimal reproduction project:** [keep-linear.zip](https://github.com/godotengine/godot/files/2948031/keep-linear.zip)

- - -

EDIT: Clarified graphics backend and reproduction.

### PR DESCRIPTION
**Pull request by ghost**
When transparent_bg is on, or the render target is too small,
Godot would skip postprocessing and disregard keep_3d_linear.

This fixes #26817.

### CODE REVIEW

File: drivers/gles3/rasterizer_scene_gles3.cpp [hunk 1]
```diff
@@ -3644,11 +3644,11 @@ void RasterizerSceneGLES3::_post_process(Environment *env, const CameraMatrix &p
 	}
 
 	if (!env || storage->frame.current_rt->flags[RasterizerStorage::RENDER_TARGET_TRANSPARENT] || storage->frame.current_rt->width < 4 || storage->frame.current_rt->height < 4) { //no post process on small render targets
-		//no environment or transparent render, simply return and convert to SRGB
+		//no environment or transparent render, simply return and convert to SRGB if KEEP_3D_LINEAR is not true.
```
Comment (reduz) [inline drivers/gles3/rasterizer_scene_gles3.cpp]: fix makes sense but i dont understand this comment, i think this was just something that was forgotten

**Review by reduz — CHANGES_REQUESTED** [review]

File: drivers/gles3/rasterizer_scene_gles3.cpp (hunk 1 above)
Comment (toasteater) [reply drivers/gles3/rasterizer_scene_gles3.cpp]: I thought it'd be good to document what the section currently does, but looking back at it the addition does seem like noise. The code is clear enough and extra verbosity isn't necessary.

Removed the change.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.1.

**Pull request merged**